### ISSUE TITLE
A key mapping depending on if there are selected files in CWD

### ISSUE THREAD

**Issue by aksr** [vifm/vifm#1037]
I was wondering if there is currently a possibility to check if there are selected files in current working directory 
and to make a mapping if that's a case.

Some of the one-character key bindings could have a *different* meaning depending on a context.
To type them fast is good and making them (a bit) *smarter* is better still.

**Thank you**, I'm *almost daily* impressed on how *powerful* `vifm` truly is.

**Comment by xaizek** [comment]
v0.14 will have `selected()` builtin function that returns number of selected files, until then you can use `expand('%c') == expand('%f')` workaround to check if more than one file is selected.  Lua API also allows enumerating entries and checking their selection status, but that's probably more complicated than what you're looking for.

**Comment by xaizek** [comment]
Aforementioned `selected()` function is available on master now (https://github.com/vifm/vifm/commit/d0a07cd3a0b8ccfdf751e72d6008e6cd2970303c).

**Comment by xaizek** [comment]
Yes, you just need to enter command-line mode as part of the mapping and use the check there.  Example:
```vim
nnoremap <silent> az : if expand('%c') == expand('%f')
                   \ |    echo 'one or no files selected'
                   \ | else
                   \ |    echo 'multiple files'
                   \ | endif<cr>
```
Running other normal commands is more complicated as they need to be wrapped to not interrupt if-statement:
```vim
nnoremap <silent> az : if expand('%c') == expand('%f')
                   \ |    execute 'normal j'
                   \ | else
                   \ |    execute 'normal k'
                   \ | endif<cr>
```
Another caveat is that most :commands reset selection, in which case [`:keepsel`](https://vifm.info/vimdoc.shtml#vifm-%3Akeepsel) can help.

**Comment by xaizek** [comment]
Version for v0.14:
```vim
nnoremap <silent> az : if selected() > 0
                   \ |    echo 'one or no files selected'
                   \ | else
                   \ |    echo 'multiple files'
                   \ | endif<cr>
```

Closing this.
